### PR TITLE
Companies API

### PR DESCRIPTION
**Pull request by abinashpa**
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have added automated tests for my code

### CODE REVIEW

File: app/controllers/internal_api/v1/companies_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+class InternalApi::V1::CompaniesController < InternalApi::V1::ApplicationController
+  def index
+    authorize current_company
+    render json: { company: current_company }
+  end
+
+  def create
+    authorize Company
+    company = Company.new(company_params)
+    if company.save
```
Comment (akhilgkrishnan) [inline app/controllers/internal_api/v1/companies_controller.rb:12]: What happen if it is not saved?

File: app/controllers/internal_api/v1/companies_controller.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      current_user.current_workspace_id = company.id
+      current_user.add_role(:owner, company)
+      current_user.save!
+      render json: { notice: I18n.t("companies.create.success") }
+    end
+  end
+
+  def update
+    authorize current_company
+    if current_company.update(company_params)
```
Comment (akhilgkrishnan) [inline app/controllers/internal_api/v1/companies_controller.rb:23]: Add the else condition

File: app/helpers/company_helper.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
```
Comment (akhilgkrishnan) [inline app/helpers/company_helper.rb:1]: remove this file, if it is not using

File: spec/helpers/company_helper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
```
Comment (akhilgkrishnan) [inline spec/helpers/company_helper_spec.rb:1]: remove this file, if it is not required

File: spec/requests/internal_api/v1/companies/create_spec.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "InternalApi::V1::Companies::Index", type: :request do
```
Comment (akhilgkrishnan) [inline spec/requests/internal_api/v1/companies/create_spec.rb:5]:
```suggestion
RSpec.describe "InternalApi::V1::Companies::Create", type: :request do
```

File: spec/requests/internal_api/v1/companies/update_spec.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "InternalApi::V1::Companies::update", type: :request do
```
Comment (akhilgkrishnan) [inline spec/requests/internal_api/v1/companies/update_spec.rb:5]:
```suggestion
RSpec.describe "InternalApi::V1::Companies::Update", type: :request do
```

File: spec/requests/internal_api/v1/companies/create_spec.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+    it "response should be successful" do
+      expect(response).to be_successful
+    end
+
+    it "returns success json response" do
+      expect(json_response["notice"]).to eq(I18n.t("companies.create.success"))
+    end
+  end
+end
```
Comment (akhilgkrishnan) [inline spec/requests/internal_api/v1/companies/create_spec.rb:34]: Add the test for the negative conditions.

File: spec/requests/internal_api/v1/companies/index_spec.rb [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+    it "response should be successful" do
+      expect(response).to be_successful
+    end
+
+    it "returns success json response" do
+      expect(json_response["company"]["name"]).to eq(company1.name)
+    end
+  end
+end
```
Comment (akhilgkrishnan) [inline spec/requests/internal_api/v1/companies/index_spec.rb:28]: Add the test for the negative conditions.

**Review by akhilgkrishnan — COMMENTED** [review]
@abinashpa Added some comments

File: app/controllers/internal_api/v1/companies_controller.rb (hunk 2 above)
Comment (abinashpa) [reply app/controllers/internal_api/v1/companies_controller.rb]: We have `concern/error_handler.rb` hence we don't need the else condition.

File: app/controllers/internal_api/v1/companies_controller.rb (hunk 1 above)
Comment (abinashpa) [reply app/controllers/internal_api/v1/companies_controller.rb]: Suppose an error is thrown then `concern/error_handler.rb` will handle that error and send the response.

File: spec/requests/internal_api/v1/companies/create_spec.rb (hunk 7 above)
Comment (akhilgkrishnan) [reply spec/requests/internal_api/v1/companies/create_spec.rb:62]: still pending

File: spec/requests/internal_api/v1/companies/create_spec.rb (hunk 5 above)
Comment (akhilgkrishnan) [reply spec/requests/internal_api/v1/companies/create_spec.rb]: still pending

File: spec/requests/internal_api/v1/companies/index_spec.rb (hunk 8 above)
Comment (akhilgkrishnan) [reply spec/requests/internal_api/v1/companies/index_spec.rb:28]: still pending

File: spec/requests/internal_api/v1/companies/update_spec.rb (hunk 6 above)
Comment (akhilgkrishnan) [reply spec/requests/internal_api/v1/companies/update_spec.rb:5]: still pending

**Review by akhilgkrishnan — APPROVED** [review]

**Review by apoorv1316 — APPROVED** [review]
